src/main/scala/service/UserService.scala

```diff
@@ -1,15 +1,22 @@
+package service
+
 import  domain.UserID._
 import dal._
 import com.github.t3hnar.bcrypt._
+import service.token.TokenService._
 
-class UserService {
+object UserService {
 
   val UserDal = new UserDal
 
-  def getUserJWT(userID:UserID, userInputtedPassword:String): Either[String,Boolean] = {
-    val (dataBasePassword,salt) = UserDal.getPasswordSalt(userID).getOrElse(("",""))
+  def getUserJWT(userID:UserID, userInputtedPassword:String): Either[String,Jwt] = {
+    val (databasePassword,salt) = UserDal.getPasswordSalt(userID).getOrElse(("",""))
+    val hashedUserInputPassword = userInputtedPassword.bcrypt(salt)
+    checkAndCreateToken(userID,hashedUserInputPassword,databasePassword)
```

Comment:
I know both parts are simple, but should check and create be done within same method?

Also as this is series of steps
-Get Password/Salt From database
-encrypt passed in password
-Check Password matches
-Create Token
Could each of these be changed to return Either[String, [Some Type]] functions and chained together with for style syntax?
This would enable each step to give a more suitable failure message i.e. Incorrect password, User Not Found, rather than  always returning "Could not make JWT"